File: saizad-mvvm/src/main/java/com/saizad/mvvm/delegation/BaseLifecycleCallBack.java
package com.saizad.mvvm.delegation;

import androidx.annotation.NonNull;

import com.saizad.mvvm.components.SaizadBaseViewModel;

public interface BaseLifecycleCallBack {

    void requestError(@NonNull SaizadBaseViewModel.ErrorData errorData);

    void requestApiError(@NonNull SaizadBaseViewModel.ApiErrorData apiErrorData);

    void requestLoading(@NonNull SaizadBaseViewModel.LoadingData loadingData);

    boolean serverError(@NonNull Throwable throwable, int requestId);

}

File: saizad-mvvm/src/main/java/com/saizad/mvvm/delegation/BaseLifecycleCallBack.kt
package com.saizad.mvvm.delegation

import com.saizad.mvvm.components.SaizadBaseViewModel.*

interface BaseLifecycleCallBack {
    fun requestError(errorData: ErrorData)
    fun requestApiError(apiErrorData: ApiErrorData)
    fun requestLoading(loadingData: LoadingData)
    fun serverError(throwable: Throwable, requestId: Int): Boolean
}